### DISCUSSION TITLE
Using markdown snippets in a repeating section based on collection tag

### DISCUSSION THREAD

**Discussion by andrewdotdobson** [11ty/eleventy#2316] [Q&A] [answered]
Hello,

I'd like to have a collection of markdown snippets in a folder in `/src` which I can render into partial templates and then fill in a repeating section in a template with them according to collection tag.

So essentially I'd have a folder of `.md` files with the format:

```
---
tags: items
title: H2 content
permalink: false
---

## {{title}}

This is some content for a repeating section.

This should render as HTML within the repeating partial.

[Link to something](link)
```

and then in my main page template be able to return a collection and  loop through each item in the collection, with it's relevant markdown content rendered appropriately:

```
{%- for item in collections.items -%}
    <h3>{{ item.data.title }}</h3>
    <div>{{ item.content }}</div>
{%- endfor -%}
```
This obviously renders the title from frontmatter, but not the content.

I've also tried doing this with a partial like this:

```
<div class="grid-item">
    <h2 class="section-headline">{{ title | safe }}<</h2>
</div>
<p>{{ content | safe }}</p>
```

And then try including it in the repeating block but this obviously doesn't work as you can't pass the relevant collection reference to the partial to tell it what to render (at least as far as my n00b brain can work out.

Is there a way of looping through a collection of posts and rendering them in a template. The .md files themselves would not need to create actual pages, I just want them converted to njk/html and inserted into a repeating section of the page so that I can maintain them easier.

Thanks.

**Comment by pdehaan** [comment, marked as answer]
I think you want `templateContent` (see https://github.com/pdehaan/11ty-2316).

```liquid
---
# src/index.liquid
title: HOME PAGE
---

<section>
{%- for item in collections.items %}
  <h3>{{ item.data.title }}</h3>
  <div>{{ item.templateContent }}</div>
{%- endfor %}
</section>
```

**Reply by pdehaan** [reply]
You can find my "fake" snippets in ./src/snippets and they generally look like this:

```md
---
title: THREE SNIPPET
---

## {{ title }}

This is some content for page three.

This should render as _fancy_ HTML.

[eleventy homepage](https://11ty.dev/)
```

Where I also have a directory data file, ./src/snippets/snippets.json, which contains all my _duplicated_ front matter:
```json
{
  "tags": ["items"],
  "permalink": false
}
```

Also note that the snippets on my output www/index.html file aren't seemingly in any predictable order (possibly by filename or something). So if you wanted snippets on the homepage in a specific order, we might have to use the collections API and create a collection manually and sort it based on date (or a custom front matter key).

**Reply by pdehaan** [reply]
Just for giggles, here's what a custom sort order via front matter keys might look like&hellip;

First add a `snippetOrder` key in each of your src/snippets/*.md files:
```diff
---
# src/snippets/three.md (snippet)
title: THREE SNIPPET
+snippetOrder: 3
---
```

Then I just added this to the .eleventy.js config file:

```js
  eleventyConfig.addCollection("items", function (collectionApi) {
    const SORT_KEY = "snippetOrder";
    const items = collectionApi.getFilteredByTag("items");
    return [...items]
      .sort((a, b) => a.data[SORT_KEY] - b.data[SORT_KEY]);
  });
```

You _shouldn't_ need any other changes. We're basically just overwriting the auto-generated `collections.items` with our sorted version. And based on my so-very-brief 2 minute testing, it seems to handle missing `snippetOrder` keys and just places those items at the bottom of the output. 🤷

**Reply by andrewdotdobson** [reply]
Amazing. thank you so much, that's bang on and I just learnt a whole load about how 11ty handles this data (and also how to make better sense of the docs, because that concept totally passed me by).  Cheers!
